### PR TITLE
MGMT-4481 Run assisted-service e2e job only if specific patterns were changed

### PR DESCRIPTION
**Pull request by YuviGold**

### CODE REVIEW

File: ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml [hunk 1]
```diff
@@ -14,6 +14,7 @@ presubmits:
       pj-rehearse.openshift.io/can-be-rehearsed: "true"
     name: pull-ci-openshift-assisted-service-master-e2e-metal-assisted
     rerun_command: /test e2e-metal-assisted
+    run_if_changed: ^(cmd/.*|deploy/.*|hack/.*|internal/.*|pkg/.*|tools/.*|Makefile|go\.mod|go\.sum)$
```
Comment (filanov) [inline ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml:17]: maybe it should be a black list instead?

Comment (YuviGold) [reply ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml:17]: Harder to follow a denylist. what's wrong with this list?

Comment (osherdp) [reply ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml:17]: prow doesn't support denylist
let's try this out, although I will feel better with ``**/*.go`` if it's possible adding to the list

Comment (YuviGold) [reply ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml:17]: But then you will run for subsystem as well

Comment (osherdp) [reply ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml:17]: yes, you're right

**Comment by filanov** [comment]
what about swagger?

**Comment by YuviGold** [comment]
> what about swagger?

swagger itself shouldn't affect, only in case the server changes its behavior. No?

**Comment by filanov** [comment]
> > what about swagger?
> 
> swagger itself shouldn't affect, only in case the server changes its behavior. No?

it should detect if someone break the API.

**Pull request merged**